src/native_loader.rs

```diff
@@ -65,26 +65,30 @@ pub fn process_transaction(keyed_accounts: &mut [KeyedAccount], tx_data: &[u8])
             }
         };
         trace!("Call native {:?}", name);
-        {
-            // create native program
-            let path = create_path(&name);
-            // TODO linux tls bug can cause crash on dlclose(), workaround by never unloading
-            let library = Library::open(Some(path), libc::RTLD_NODELETE | libc::RTLD_NOW).unwrap();
-            unsafe {
-                let entrypoint: Symbol<Entrypoint> = match library.get(ENTRYPOINT.as_bytes()) {
-                    Ok(s) => s,
-                    Err(e) => {
-                        warn!("{:?}: Unable to find {:?} in program", e, ENTRYPOINT);
-                        return false;
-                    }
-                };
-                return entrypoint(&mut keyed_accounts[1..], tx_data);
-            }
+        let path = create_path(&name);
+        // TODO linux tls bug can cause crash on dlclose(), workaround by never unloading
+        let library = Library::open(Some(path), libc::RTLD_NODELETE | libc::RTLD_NOW).unwrap();
+        unsafe {
+            let entrypoint: Symbol<Entrypoint> = match library.get(ENTRYPOINT.as_bytes()) {
+                Ok(s) => s,
+                Err(e) => {
+                    warn!("{:?}: Unable to find {:?} in program", e, ENTRYPOINT);
+                    return false;
+                }
+            };
+            return entrypoint(&mut keyed_accounts[1..], tx_data);
         }
     } else if let Ok(instruction) = deserialize(tx_data) {
         match instruction {
             LoaderInstruction::Write { offset, bytes } => {
                 trace!("NativeLoader::Write offset {} bytes {:?}", offset, bytes);
+                if keyed_accounts[0].account.executable {
```

Comment:
This check can be removed right?  The bank should enforce that the account is read-only

Reply:
How do you propose the loader determine if this is a dispatch or a write/finalize?  (This is where call came in handy)

Reply:
Sounds good, is that enforcement already in the bank?

Reply:
I don't think it is, sry!